frontend/src/js/controller.profileController.js

```diff
@@ -1,8 +1,15 @@
 (function() {
     angular.module("App")
-    .controller("profileController", ["User", "Countries",
-                                function(User, Countries) {
+    .controller("profileController", ["User", "Countries", "SFTPAccess", 
+                              function(User,   Countries,   SFTPAccess) {
     var localThis = this;
+    localThis.sftp = {"user":"", "password":"", "expires":null};
+    localThis.createSFTPCredentials = function() {
```

Comment:
Declare the function at the end of the controller (but assign it to localThis here). Look at how it's done in `datasetDownloadController.js`.